napari/components/dims.py

```diff
@@ -218,16 +218,33 @@ def _check_dims(cls, values):
         elif labels_ndim > ndim:
             updated['axis_labels'] = axis_labels[-ndim:]
 
+        # If the last used slider is no longer visible, use the first.
+        last_used = values['last_used']
+        ndisplay = values['ndisplay']
+        dims_range = values['range']
+        not_displayed = [
+            d
+            for d in order[:-ndisplay]
+            if len(cls._nsteps_from_range(dims_range)) > d
```

Comment:
Optional: mild preference for pulling out `nsteps = cls._nsteps_from_range(dims_range)` because it avoids a little extra compute and is slightly easier to read non-nested/chained expressions.